external-gateways/stellar/src/lib.rs

```diff
@@ -75,19 +75,19 @@ impl TryFrom<Message> for ScVal {
 
     fn try_from(value: Message) -> Result<Self, XdrError> {
         let keys: [&'static str; 5] = [
-            "contract_address",
+            "source_chain",
             "message_id",
-            "payload_hash",
             "source_address",
-            "source_chain",
+            "contract_address",
+            "payload_hash",
```

Comment:
note that this doesn't start with `source_chain`. I believe that's because XDR sorts the keys first, so the order within the type doesn't matter. Add a golden test in Soroban for the message XDR encoding to compare. Pretty sure this shouldn't be changed in this case (add a comment on the behaviour if it's sorting to clarify this in the future)

Reply:
Added comment, updated.